hr_public_holidays/hr_public_holidays.py

```diff
@@ -127,32 +121,21 @@ def get_holidays_list(self, cr, uid, year, employee_id=None,
                                employee.address_id.state_id.id)
```

Comment:
This whole block can be simplified a lot (from line 115 to 121)
It can be replaced with one line.
states_filter.append('state_ids', 'in', list({False, employee.address_id.state_id.id})